File: huffman_c++/Itm.cpp
// Itm.cpp
// C++ version of Itm decoder for HuffmanCodeTree
#include <iostream>
#include <string>
#include "HuffmanCodeTree.cpp"
using namespace std;

class Itm {
public:
    string decoder(HuffmanCodeTree& huffmanCodeTree, const string& encodedString) {
        string output;
        HNode* current = huffmanCodeTree.root;
        for (char c : encodedString) {
            if (!current) return "";
            if (c == '0') current = current->left;
            else if (c == '1') current = current->right;
            if (current && !current->left && !current->right) {
                output += current->character;
                current = huffmanCodeTree.root;
            }
        }
        return output;
    }
};

int main() {
    cout << "Itm C++ version decoder for HuffmanCodeTree." << endl;
    // Example usage:
    // HuffmanCodeTree huffmanTree;
    // string encoded = "101100101";
    // Itm itm;
    // string decoded = itm.decoder(huffmanTree, encoded);
    // cout << "Decoded string: " << decoded << endl;
    return 0;
}

File: huffman/huffman/Itm.java
package huffman;
public class Itm{
    public String decoder(HuffmanCodeTree huffmanCodeTree,String encodedString){
        //TenaryTreeSolution in Iterative Format
        //using while?
        //Using Recursive?
        //output+=current.left.character;  return nothing
        //System.out.print(encodedString) infinite loop
        //How tree is constructed? If we input abc?,aad?
        String output="";
        HNode current=huffmanCodeTree.root;
    for(int i=0;i<encodedString.length();i++){
        if(current==null){
            return""; //return ""?
        }
        if(current.left!='\0'){ //if(current.left!='\0'){ operator can't be applied
//recursive call by another function
        }
        if(current.right!="\0"){

        }
        if(current.left==null&&current.right==null){
            output+=current.character;
        }

    }
    return output;
    }
}